lib/rules/define-macros-order.js

```diff
@@ -45,7 +57,11 @@ function getTargetStatementPosition(program) {
   ])
 
   for (const [index, item] of program.body.entries()) {
-    if (!skipStatements.has(item.type) && !isUseStrictStatement(item)) {
+    if (
+      inScriptSetup(scriptSetup, item) &&
```

Comment:
Script setup stuff fixes multiple scripts bug(test is here: https://github.com/vuejs/eslint-plugin-vue/pull/1861/files#diff-6ebdd3c21be70e29ce94a2f399884fd220d4fe9bba358ade66b0bceb325d3102R387)